### PR TITLE
chore(example): added WASM support to example

### PR DESCRIPTION
**Pull request by JaffaKetchup**
Supersedes #1867 by @josxha.

- Reset web index.html config file to initial state in Flutter 3.22
- Also adds required headers to Firebase config (hopefully)

Would like to add an indicator using dart-lang/sdk/issues/55694 when it becomes available, to confirm usage of WASM.

### CODE REVIEW

**Comment by JaffaKetchup** [comment]
> It's possible to deploy branches temporarily on firebase hosting, see [FirebaseExtended/action-hosting-deploy#channelid-string](https://github.com/FirebaseExtended/action-hosting-deploy?tab=readme-ov-file#channelid-string).

Yep, I've looked into this extensively before. However, I disabled it, as most of our PRs come from forks, and for security reasons, GitHub Actions and Firebase won't do this from a fork PR. I can try deploying a preview version manually perhaps?

**Comment by JaffaKetchup** [comment]
https://fleaflet-firebase--preview-wasm-bfczu5tt.web.app/

Can confirm it doesn't work, but only because the CTP uses Dio which doesn't support Dio yet. Not sure why I couldn't reproduce this locally, but I guess we'll have to hold off for now.

**Comment by josxha** [comment]
Open pull request here:
- https://github.com/cfug/dio/pull/2215

**Comment by JaffaKetchup** [comment]
Waiting for:

- https://github.com/cfug/dio/pull/2215
- A Flutter release (beta channel) with Dart [3.5.0-185.0.dev](https://github.com/dart-lang/sdk/releases/tag/3.5.0-185.0.dev), to include https://github.com/dart-lang/sdk/commit/6b71aa1f3e997fcbbb19930a1c029ffcebf33d67

**Comment by lishaduck** [comment]
Since [web_adapter v2.0.0](https://github.com/cfug/dio/releases/tag/web_adapter_v2.0.0), dio shouldn't be a blocker anymore.
Dart 3.5 is out as of Flutter 3.24.

**Comment by JaffaKetchup** [comment]
Wasm is now available! See it working in the preview (confirm with the drawer): https://fleaflet-firebase--wasm-test-g7fo47s3.web.app/.

File: example/lib/widgets/drawer/menu_drawer.dart [hunk 1]
```diff
@@ -61,6 +63,11 @@ class MenuDrawer extends StatelessWidget {
                   textAlign: TextAlign.center,
                   style: TextStyle(fontSize: 14),
                 ),
+                const Text(
+                  _isWASM ? 'Running with WASM' : 'Running without WASM',
+                  textAlign: TextAlign.center,
+                  style: TextStyle(fontSize: 14),
+                ),
```
Comment (josxha) [inline example/lib/widgets/drawer/menu_drawer.dart:70]: idea: only display this text when running on web, could cause confusion when running on other platforms.

Comment (JaffaKetchup) [reply example/lib/widgets/drawer/menu_drawer.dart]: Good idea, done.

**Review by josxha — APPROVED** [review]

**Pull request merged**